eth/catalyst/api_test.go

```diff
@@ -163,6 +164,8 @@ func TestEth2PrepareAndGetPayload(t *testing.T) {
 	if err != nil {
 		t.Fatalf("error preparing payload, err=%v", err)
 	}
+	// give the payload some time to be built
+	time.Sleep(100 * time.Millisecond)
```

Comment:
I tested this instead, and kind of expected to 'see' the build-up from an empty block to a full block: 
```golang	//time.Sleep(100 * time.Millisecond)
	payloadID := computePayloadId(fcState.HeadBlockHash, &blockParams)
	for {
		execData, err := api.GetPayloadV1(payloadID)
		if err != nil {
			t.Fatalf("error getting payload, err=%v", err)
		}
		if len(execData.Transactions) != blocks[9].Transactions().Len() {
			t.Logf("invalid number of transactions %d != 1", len(execData.Transactions))
		} else {
			break
		}
		time.Sleep(15 * time.Millisecond)
	}
```
However, it just looped on `    api_test.go:176: invalid number of transactions 0 != 1` and didn't produce anything. What am I doing wrong?

Reply:
Once `api.GetPayloadV1` is called, the background block generator is stopped and the current available block is returned as the final payload, no matter it's full or empty. 

In your case, since you call `GetPayloadV1 ` without any time sleep, so the empty block is retrieved back.